UrlManager.php

```diff
@@ -370,6 +382,14 @@ protected function processLocaleUrl($normalized)
             if ($this->enableLanguagePersistence) {
                 $language = $this->loadPersistedLanguage();
             }
+            if ($language===null && isset($_SERVER[$this->geoipServerFieldName]) && !empty($this->geoipLanguageCountries)) {
+                foreach ($this->geoipLanguageCountries as $key => $codes) {
+                    if (in_array($_SERVER[$this->geoipServerFieldName], $codes)) {
+                        $language = $key;
+                        break;
+                    }
+                }
+            }
```

Comment:
Browser preferences should always have highest priority. So this block should be moved further down.

Reply:
i was thinking about placing it after enableLanguageDetection block
but detecting by geoip and enableLanguageDetection are excluding each other a bit
and it looks like it does not make sense to detect by geoip if you have enableLanguageDetection set to true and it has higher priority

Reply:
Hmm, I don't think that they exclude each other. If someone has configured for example "en" in the browser then the user has made a clear decision. He knows, which language he wants and he tells us via the header fields. So even if this user is in holiday in some foreign country, we should not suddenly show him a different language.

So all there is to do is really to move this down a little bit, then everything should be fine.

Reply:
ok, changed everything as you suggested